packages/compiler-vapor/src/transforms/transformSlotOutlet.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  type AttributeNode,
+  type DirectiveNode,
+  type ElementNode,
+  ElementTypes,
+  ErrorCodes,
+  NodeTypes,
+  type SimpleExpressionNode,
+  createCompilerError,
+  createSimpleExpression,
+  isStaticArgOf,
+  isStaticExp,
+} from '@vue/compiler-core'
+import type { NodeTransform, TransformContext } from '../transform'
+import {
+  type BlockIRNode,
+  DynamicFlag,
+  IRNodeTypes,
+  type IRProps,
+  type VaporDirectiveNode,
+} from '../ir'
+import { camelize, extend, isBuiltInDirective } from '@vue/shared'
+import { newDynamic } from './utils'
+import { buildProps } from './transformElement'
+
+export const transformSlotOutlet: NodeTransform = (node, context) => {
+  if (node.type !== NodeTypes.ELEMENT || node.tag !== 'slot') {
+    return
+  }
+  const { props } = node
+  const id = context.reference()
+  context.dynamic.flags |= DynamicFlag.INSERT
+  const [fallback, exitBlock] = createFallback(
+    node,
+    context as TransformContext<ElementNode>,
+  )
+
+  let name: SimpleExpressionNode | undefined
+  const nonNameProps: (AttributeNode | DirectiveNode)[] = []
+  const customDirectives: DirectiveNode[] = []
+  for (const p of props) {
+    if (p.type === NodeTypes.ATTRIBUTE) {
+      if (p.value) {
+        if (p.name === 'name') {
+          name = createSimpleExpression(p.value.content, true, p.loc)
+        } else {
+          p.name = camelize(p.name)
```

Comment:
AST should not be mutated. It'll break cache.

Reply:
Maybe [this code](https://github.com/vuejs/core-vapor/blob/d3a1df222a0f1009ab26b97c8670a39a36f559db/packages/compiler-vapor/src/transforms/vBind.ts#L61) has a similar problem, do we need to fix it together too?

I may lack knowledge about this caching mechanism, can you briefly describe it? Or how about I can learn about it? Many thanks.

Reply:
Mentioned before https://github.com/vuejs/core-vapor/pull/98#discussion_r1468877535, see https://github.com/vuejs/core-vapor/commit/d3baff92b1fb033a1411223a009a8bf68c432037


Cause AST will be reused at the second usage, and could be transformed twice.
https://github.com/vuejs/core-vapor/blob/2d684293f2bd0facf882b1aed126ada5553a2660/packages/compiler-sfc/src/compileTemplate.ts#L226

Reply:
The case you mentioned should be fixed.